pkg/controller/daemon/daemoncontroller.go

```diff
@@ -548,10 +560,12 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) error {
 	glog.V(4).Infof("Nodes needing daemon pods for daemon set %s: %+v, creating %d", ds.Name, nodesNeedingDaemonPods, createDiff)
 	createWait := sync.WaitGroup{}
 	createWait.Add(createDiff)
+	dsPodTemplateSpecHash := podutil.GetPodTemplateSpecHashFnv(ds.Spec.Template)
+	template := util.GetPodTemplateWithHash(ds, dsPodTemplateSpecHash)
```

Comment:
Suggest putting the "calculating hash" part inside `GetPodTemplateWithHash` and we only pass in `ds.Spec.Template`